src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8216,28 +8230,55 @@ internal static bool IsReparsePoint(FileSystemInfo fileInfo)
             return fileInfo.Attributes.HasFlag(System.IO.FileAttributes.ReparsePoint);
         }
 
-        internal static bool IsReparsePointWithTarget(FileSystemInfo fileInfo)
+        internal static bool IsReparsePointLikeSymlink(FileSystemInfo fileInfo)
         {
-            if (!IsReparsePoint(fileInfo))
+#if UNIX
+            // Reparse point on Unix is a symlink.
+            return IsReparsePoint(fileInfo);
+#else
+            if (InternalTestHooks.OneDriveTestOn && fileInfo.Name == InternalTestHooks.OneDriveTestSymlinkName)
             {
-                return false;
+                return !InternalTestHooks.OneDriveTestRecurseOn;
             }
-#if !UNIX
-            // It is a reparse point and we should check some reparse point tags.
-            var data = new WIN32_FIND_DATA();
-            using (var handle = FindFirstFileEx(fileInfo.FullName, FINDEX_INFO_LEVELS.FindExInfoBasic, ref data, FINDEX_SEARCH_OPS.FindExSearchNameMatch, IntPtr.Zero, 0))
+
+            WIN32_FIND_DATA data = default;
+            string fullPath = Path.TrimEndingDirectorySeparator(fileInfo.FullName);
+            if (fullPath.Length > MAX_PATH)
```

Comment:
Personally I think we should always just ensure the extended prefix is set. This means you are always testing the same type of path and if there is any faulty logic there it will always be apparent and fixed up rather than hidden behind a less common scenario.

Reply:
This would be perf regression.
Here we follow .Net Runtime practice.

Reply:
Performance regression in the sense we concat a string or some other reason? I'm very new to the .NET world when it comes to performance tuning so I'm surprised that this would be considered a performance problem.

Reply:
Performance has several varieties:
1. External API delays (like IO waiting)
2. High CPU usage
3. High memory allocations with GC pressure.

If you did run an profiler you could see how many allocations PowerShell does. 
We should avoid extra allocations.
This is the case here.

---
Generally speaking, PowerShell only supports long paths because .Net does. So you won't find our code adding extended prefixes anywhere.
So it's advantageous for us to migrate to .Net API wherever possible, because this API is always implemented with long path support in at its best perf.